apps/myaccount/src/components/shared/system-notification-alert.tsx

```diff
@@ -0,0 +1,207 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Alert from "@oxygen-ui/react/Alert";
+import { ProfileConstants } from "@wso2is/core/constants";
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import {  LinkButton } from "@wso2is/react-components";
+import { AxiosError } from "axios";
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { Dispatch } from "redux";
+import { getPreference, resendVerificationLinkOrCode } from "../../api";
+import { ProfileConstants as MyAccountProfileConstants, RecoveryScenario } from "../../constants";
+import { SCIMConfigs } from "../../extensions/configs/scim";
+import { AlertLevels, AuthStateInterface } from "../../models";
+import { PreferenceConnectorResponse, PreferenceProperty, PreferenceRequest } from "../../models/preference";
+import { AppState } from "../../store";
+import { addAlert } from "../../store/actions";
+import "./system-notification-alert.scss";
+
+/**
+ * SystemNotificationAlert component props.
+ */
+export type SystemNotificationAlertProps = IdentifiableComponentInterface;
+
+/**
+ * System Notification Alert component.
+ *
+ * @param props - Props injected into the component.
+ * @returns ReactElement
+ */
+export const SystemNotificationAlert: FunctionComponent<SystemNotificationAlertProps> = (
+    props: SystemNotificationAlertProps
+): ReactElement | null => {
+    const { [ "data-componentid" ]: componentId } = props;
+
+    const { t } = useTranslation();
+    const dispatch: Dispatch = useDispatch();
+
+    const profileDetails: AuthStateInterface = useSelector((state: AppState) => state.authenticationInformation);
+
+    const [ isSRSendOtpInEmailEnabled, setIsSRSendOtpInEmailEnabled ] = useState<boolean>(false);
+    const [ isAccountStatePendingSR, setIsAccountStatePendingSR ] = useState<boolean>(false);
+    const [ isPreferredChannelEmail, setIsPreferredChannelEmail ] = useState<boolean>(false);
+    const [ isSubmitting, setIsSubmitting ] = useState<boolean>(false);
+
+    useEffect(() => {
+        getSelfSignUpPreferences();
+    }, []);
+
+    useEffect(() => {
+        const accountState: string = profileDetails
+            ?.profileInfo
+            ?.[SCIMConfigs?.scim?.systemSchema]
+            ?.[ProfileConstants.SCIM2_SCHEMA_DICTIONARY.get(MyAccountProfileConstants.ACCOUNT_STATE)];
+
+        const preferredChannel: string = profileDetails
+            ?.profileInfo
+            ?.[SCIMConfigs?.scim?.systemSchema]
+            ?.[ProfileConstants.SCIM2_SCHEMA_DICTIONARY.get(MyAccountProfileConstants.PREFERRED_CHANNEL)];
+
+        setIsAccountStatePendingSR(accountState === MyAccountProfileConstants.ACCOUNT_STATE_PENDING_SR);
```

Comment:
what SR mean here is ambigious, IMO. shall we improve the variable naming?

Reply:
Resolved via https://github.com/wso2/identity-apps/pull/8071/commits/ad689a3e1bfb488e07f08dbac7018befc1241bcc